code_generation/license.h

```diff
@@ -83,6 +87,27 @@ class KODE_EXPORT License
      */
     QString text() const;
 
+    /**
+     * Returns license Type enum element by name. If there no such license,
+     * NoLicense will be returned
+     * @param license_string Name of license
+     * @return one of Type elements
+     */
+    static Type getLicenseNameByString(const QString &license_string);
+
+    /**
+     * Return license name by Type enum element.
+     * @param license_enum one of Type elements
+     * @return name of license
+     */
+    static QString getLicenseNameByEnum(const Type &license_enum);
+
+    /**
+     * Returns list of supported licenses
+     * @return space delimited license names
+     */
+    static QString getSupportedLicenses();
```

Comment:
It's weird for this to return one big QString. Shouldn't it be a QVector<Type> instead? or even QVector<License>.... That's much more typesafe and more generally usable than a space-separated string.

Reply:
I decided to change signature to `static QStringList getSupportedLicenses()`